pkg/repository/maintenance.go

```diff
@@ -262,3 +141,89 @@ func getLatestMaintenanceJob(cli client.Client, ns string) (*batchv1.Job, error)
 
 	return &jobList.Items[0], nil
 }
+
+// getMaintenanceJobConfig is called to get the Maintenance Job Config for the
+// BackupRepository specified by the repo parameter.
+//
+// Params:
+//
+//	ctx: the Go context used for controller-runtime client.
+//	client: the controller-runtime client.
+//	logger: the logger.
+//	veleroNamespace: the Velero-installed namespace. It's used to retrieve the BackupRepository.
+//	repoMaintenanceJobConfig: the repository maintenance job ConfigMap name.
+//	repo: the BackupRepository needs to run the maintenance Job.
+func getMaintenanceJobConfig(
+	ctx context.Context,
+	client client.Client,
+	logger logrus.FieldLogger,
+	veleroNamespace string,
+	repoMaintenanceJobConfig string,
+	repo *velerov1api.BackupRepository,
+) (*JobConfigs, error) {
+	var cm v1.ConfigMap
+	if err := client.Get(
+		ctx,
+		types.NamespacedName{
+			Namespace: veleroNamespace,
+			Name:      repoMaintenanceJobConfig,
+		},
+		&cm,
+	); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, nil
+		} else {
+			return nil, errors.Wrapf(
+				err,
+				"fail to get repo maintenance job configs %s", repoMaintenanceJobConfig)
+		}
+	}
+
+	if cm.Data == nil {
+		return nil, errors.Errorf("data is not available in config map %s", repoMaintenanceJobConfig)
+	}
+
+	configMap := make(map[string]JobConfigs)
+	for k, v := range cm.Data {
+		var config JobConfigs
+		if err := json.Unmarshal([]byte(v), &config); err != nil {
+			return nil, errors.Wrapf(err, "fail to unmarshal configs from %s", repoMaintenanceJobConfig)
+		}
+		configMap[k] = config
+	}
+
+	// Generate the BackupRepository key.
+	// If using the BackupRepository name as the is more intuitive,
+	// but the BackupRepository generation is dynamic. We cannot assume
+	// they are ready when installing Velero.
+	// Instead we use the volume source namespace, BSL name, and the uploader
+	// type to represent the BackupRepository. The combination of those three
+	// keys can identify a unique BackupRepository.
+	repoJobConfigKey := repo.Spec.VolumeNamespace + "-" +
+		repo.Spec.BackupStorageLocation + "-" + repo.Spec.RepositoryType
+
+	var result *JobConfigs
+	if _, ok := configMap[repoJobConfigKey]; ok {
+		logger.Debugf("Find the repo maintenance config %s for repo %s", repoJobConfigKey, repo.Name)
+		result = new(JobConfigs)
+		*result = configMap[repoJobConfigKey]
+	}
+
+	if _, ok := configMap["global"]; ok {
+		logger.Debugf("Find the global repo maintenance config for repo %s", repo.Name)
+
+		if result == nil {
+			result = new(JobConfigs)
+		}
+
+		if result.PodResources == nil && configMap["global"].PodResources != nil {
+			result.PodResources = configMap["global"].PodResources
+		}
+
+		if len(result.LoadAffinities) == 0 {
+			result.LoadAffinities = configMap["global"].LoadAffinities
+		}
+	}
```

Comment:
Perhaps only Unmarshal as needed?

This is Unmarshaling on everything in configmap.data then only using one of the result.

Reply:
Fixed.